### PR TITLE
Feature/add shape data

### PR DESCRIPTION
**Pull request by rockwood**
This also fixes some issue with the importer. This shape data isn't currently isn't being used, but I'd like to get this out with the importer fixes so we can get the updated GTFS data. @geofflane could you take a look?

### CODE REVIEW

File: app/apis/metro/importer.rb [hunk 1]
```diff
@@ -31,28 +31,29 @@ def import!
...
+    @agency.services.where("services.remote_id NOT IN (?)", source.calendars.map(&:service_id)).destroy_all
+    GC.start
   end
 
   def import_services!
-    source.each_calendar do |cal|
-      service = Service.find_or_initialize_by(remote_id: cal.service_id, agency: @agency)
-      service.attributes = {
+    source.calendars.each do |cal|
+      Service.find_or_initialize_by(remote_id: cal.service_id, agency: @agency).update!(
```
Comment (geofflane) [inline app/apis/metro/importer.rb]: Are we sure that find_or_initialize_by..update! will do an INSERT on a new record?

File: app/models/shape_point.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class ShapePoint < ActiveRecord::Base
+  belongs_to :shape
+  has_one :agency, through: :shape
```
Comment (geofflane) [inline app/models/shape_point.rb]: Do we really need the ShapePoint.agency? has_one :through doesn't give us anything at the data layer, so I'm just wondering if we use that anywhere?

File: app/models/trip_searcher.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    @params = params
+    @per_page = positive_or_default(params[:per_page].to_i, DEFAULT_PER_PAGE)
+    @page = positive_or_default(params[:page].to_i, 1)
+  end
+
+  def results
+    @results ||= paginated_results
+  end
+
+  def total_results
```
Comment (geofflane) [inline app/models/trip_searcher.rb]: We should use Kaminari for this stuff. It makes all of that a lot easier. I think @tmecklem integrated Kaminari on another branch already.

File: app/models/trip_searcher.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+  def valid?
+    [
+      @params[:route_id].present?,
+    ].any?
+  end
+
+  private
+
+  def positive_or_default(v, d)
```
Comment (geofflane) [inline app/models/trip_searcher.rb]: We should extract this into a Searchable concern or something similar. We've got a lot of duplication between this and the StopSearcher that could probably be put in there as well.

File: app/serializers/trip_searcher_serializer.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class TripSearcherSerializer < ApplicationSerializer
```
Comment (geofflane) [inline app/serializers/trip_searcher_serializer.rb]: We could introduce a base SearcherSerializer and inherit from that for Trips and Stops. They both would then have the exact same APIs.

**Comment by geofflane** [comment]
Mainly just a few refactoring chores to clean some things up I think. Nothing major sticks out @rockwood. But we can merge this and cleanup later so we can get the latest data in tonight.

File: app/models/shape_point.rb (hunk 2 above)
Comment (rockwood) [inline app/models/shape_point.rb]: It's just so we can do `@agency.stop_points.destroy` in the importer. Looked a little better than joining through shapes within the importer

**Comment by rockwood** [comment]
Totally agree about a using Kaminari and extracting search into shared logic. I just wanted to wait for @tmecklem's branch before doing that.

**Comment by rockwood** [comment]
Check out the source for `update`. Looks like all it does is calls `set_attributes` and `save`.

http://apidock.com/rails/ActiveRecord/Persistence/update

**Comment by rockwood** [comment]
@geofflane any ideas on how we can limit memory usage during import?

**Comment by geofflane** [comment]
:+1:

**Comment by geofflane** [comment]
@rockwood Not off hand on the memory usage. I wonder if it's because it's in one transaction?

**Pull request merged**